internal/pkg/cli/dependencies/repository.go

```diff
@@ -68,12 +70,72 @@ func (v *container) repositoryFs(definition model.TemplateRepository, template m
 		}
 		return aferofs.NewLocalFs(v.Logger(), path, definition.WorkingDir)
 	case model.RepositoryTypeGit:
-		return git.CheckoutTemplateRepositoryPartial(template, v.Logger())
+		return gitRepositoryFs(definition, template, v.Logger())
 	default:
 		panic(fmt.Errorf(`unexpected repository type "%s"`, definition.Type))
 	}
 }
 
+// gitRepositoryFs returns template repository FS, which has been loaded from a git repository.
+// Sparse checkout is used to load only the needed files.
+func gitRepositoryFs(definition model.TemplateRepository, tmplRef model.TemplateRef, logger log.Logger) (filesystem.Fs, error) {
+	// Checkout Git repository in sparse mode
+	gitRepository, err := git.Checkout(definition.Url, definition.Ref, true, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	// Clear temp directory at the end.
+	// Files will be copied to memory.
+	defer gitRepository.Clear()
+
+	// Load repository manifest
+	if err := gitRepository.Load(".keboola/repository.json"); err != nil {
+		return nil, err
+	}
+	repoManifest, err := repository.LoadManifest(gitRepository.Fs())
+	if err != nil {
+		return nil, err
+	}
+
+	// Get version record
+	version, err := repoManifest.GetVersion(tmplRef.TemplateId(), tmplRef.Version())
+	if err != nil {
+		// version or template not found
+		e := utils.NewMultiError()
+		e.Append(fmt.Errorf(`searched in git repository "%s"`, gitRepository.Url()))
+		e.Append(fmt.Errorf(`reference "%s"`, gitRepository.Ref()))
+		return nil, utils.PrefixError(err.Error(), e)
+	}
+
+	// Load template src directory
+	srcDir := filesystem.Join(version.Path(), template.SrcDirectory)
+	if err := gitRepository.Load(srcDir); err != nil {
+		return nil, err
+	}
+	if !gitRepository.Fs().Exists(srcDir) {
+		e := utils.NewMultiError()
+		e.Append(fmt.Errorf(`searched in git repository "%s"`, gitRepository.Url()))
+		e.Append(fmt.Errorf(`reference "%s"`, gitRepository.Ref()))
+		return nil, utils.PrefixError(fmt.Sprintf(`folder "%s" not found`, srcDir), e)
+	}
+
+	// Load common directory, shared between all templates in repository, if it exists
+	if err := gitRepository.Load(template.CommonDirectory); err != nil {
+		return nil, err
+	}
```

Comment:
Rovno som tu pridal, ze sa stiahne aj `_common` adresar, ak existuje, je to aj v testoch dalej.
Je to priprava pre featuru co potrebuje Monika, ale je to iba cast.